src/benchmark/benchmark_playground.cpp

```diff
@@ -0,0 +1,62 @@
+#include <memory>
+
+#include "benchmark/benchmark.h"
+#include "benchmark_basic_fixture.hpp"
+
+namespace opossum {
+
+using ValueT = int32_t;
+
+class BenchmarkPlaygroundFixture : public BenchmarkBasicFixture {
+ public:
+  void SetUp(::benchmark::State& state) override {
+    BenchmarkBasicFixture::SetUp(state);
+
+    _vec.resize(1'000'000);
+    std::generate(_vec.begin(), _vec.end(), []() {
+      static ValueT v = 0;
+      v = ++v % 4;
+      return v;
+    });
+  }
+  void TearDown(::benchmark::State& state) override { BenchmarkBasicFixture::TearDown(state); }
+
+ protected:
+  std::vector<ValueT> _vec;
+};
+
+BENCHMARK_F(BenchmarkPlaygroundFixture, BM_Playground_Reference)(benchmark::State& state) {
+  clear_cache();
+
+  while (state.KeepRunning()) {
+    std::vector<size_t> result;
+    benchmark::DoNotOptimize(result.data());
```

Comment:
I would have expected this to be at the end of the benchmark when `result.data()` actually contains content

Reply:
Unfortunately, (as far as I understand this correctly), this is not what `benchmark::DoNotOptimize(e)` does. It just marks this memory as globally aliased.
https://stackoverflow.com/questions/40122141/ has a nice explanation.